src/vcf/validator.cpp

```diff
@@ -150,54 +162,113 @@ namespace ebi
             return false;
         }
         std::unique_ptr<Parser> validator = build_parser(sourceName, validationLevel, version);
-        return validate(line, input, *validator, outputs);
+        return validate(line, uncompressed_input, *validator, outputs);
     }
 
-    bool is_compressed_file(const std::string &source,
-                            const std::vector<char> &line)
+    void create_uncompressed_stream(std::istream & input,
+                                    const std::string & sourceName,
+                                    boost::iostreams::filtering_istream & uncompressed_input)
     {
-        if (source != ebi::vcf::STDIN && is_compressed_extension(source)) {
-            return true;
+        std::vector<char> line;
+        get_magic_num(input, line);
+
+        std::string file_ext = get_compression(sourceName, line);
+        check_readability_of_file(file_ext);
+
+        if(file_ext == BZ2) {
```

Comment:
I saw boost provides also `boost::iostreams::zlib_decompressor`, maybe we can add `.zip` as well to readable files?